### PR TITLE
Upgrade csi sidecars to newest versions on rc

### PR DESCRIPTION
**Pull request by Jiawei0227**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Upgrade csi sidecars to newest versions on staging rc for tests

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/cc @saad-ali 
/cc @verult 
/cc @saikat-royc

### CODE REVIEW

**Comment by saikat-royc** [comment]
It will be good to update the rbac for csi resizer to enable pod watch list get as here:
https://github.com/kubernetes-csi/external-resizer/blob/release-1.0/deploy/kubernetes/rbac.yaml

If we want to use the 'handle-volume-inuse-error' flag to true in future

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+- op: remove
+  path: /spec/template/spec/containers/0/args/5
+
+- op: remove
```
Comment (saikat-royc) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml:4]: can you put a comment on the arg being removed?

Comment (Jiawei0227) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml:4]: Done

File: deploy/kubernetes/base/controller/cluster_setup.yaml [hunk 2]
```diff
@@ -139,6 +146,9 @@ rules:
   - apiGroups: [""]
     resources: ["events"]
     verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
```
Comment (saikat-royc) [inline deploy/kubernetes/base/controller/cluster_setup.yaml:149]: we can be consistent in the comments about why a specific resource verbs is added (i.e here if handle-volume-inuse-error=true, the pod specific rbac is needed). For e.g as you have explained volumeattachments for provisioner, we can put a small comment here. Or we can remove from both the places.

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# "--leader-election-type=leases"
```
Comment (saikat-royc) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml:1]: also mention the container name

**Review by saikat-royc — COMMENTED** [review]
The changes w.r.t sidecar versions and rbac looks good to me. Minor comments about  comments :)

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml (hunk 3 above)
Comment (Jiawei0227) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/sidecar-upgrade.yaml:2]: Done

File: deploy/kubernetes/base/controller/cluster_setup.yaml (hunk 2 above)
Comment (Jiawei0227) [reply deploy/kubernetes/base/controller/cluster_setup.yaml:150]: Done

Comment (verult) [reply deploy/kubernetes/base/controller/cluster_setup.yaml:150]: Is this a resizer flag? Could you mentioned that in the comment here?

**Pull request merged**